File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();

        for (int i = 0; i < s.length(); i++) {
            char alpha = s.charAt(i);

            switch(alpha) {
                case '{':   stack.push(alpha);
                            break;

                case '[':   stack.push(alpha);
                            break;

                case '(':   stack.push(alpha);
                            break;

                case '}':   if (stack.empty()) {
                                return false;
                            }
                            if (stack.peek() == '{') {
                                stack.pop();
                            } else {
                                return false;
                            }
                            break;

                case ']':   if (stack.empty()) {
                                return false;
                            }
                            if (stack.peek() == '[') {
                                stack.pop();
                            } else {
                                return false;
                            }
                            break;

                case ')':   if (stack.empty()) {
                                return false;
                            }
                            if (stack.peek() == '(') {
                                stack.pop();
                            } else {
                                return false;
                            }
                            break;
                
                default: return false;
            }
        }

        if (stack.empty()) {
            return true;
        } else {
            return false;
        }
    }
}

File: Data Structures & Algorithms/minimum-stack/submission-1.kt
class MinStack() {
    val stack = Stack<Int>()
    val minStack = Stack<Int>()

    fun push(`val`: Int) {
        stack.push(`val`)
        if (minStack.size == 0) {
            minStack.push(`val`)
        } else {
            if (minStack.peek() <= `val`) {
                minStack.push(minStack.peek())
            } else {
                minStack.push(`val`)
            }
        }
    }

    fun pop() {
        stack.pop()
        minStack.pop()
    }

    fun top(): Int {
        return stack.peek()
    }

    fun getMin(): Int {
        return minStack.peek()
    }
}
